ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -146,25 +145,23 @@ StatusDialog {
                 }
 
                 currencyStore: root.swapAdaptor.currencyStore
-                flatNetworksModel: root.swapAdaptor.filteredFlatNetworksModel
-                processedAssetsModel: root.swapAdaptor.processedAssetsModel
+                flatNetworksModel: root.swapAdaptor.swapStore.flatNetworks
+                processedAssetsModel: root.swapAdaptor.walletAssetsStore.groupedAccountAssetsModel
 
                 tokenKey: root.swapInputParamsForm.fromTokensKey
-                tokenAmount: {
-                    // Only update if there is different in amount displayed
-                    if (root.swapInputParamsForm.fromTokenAmount !==
-                            SQUtils.AmountsArithmetic.fromString(value).toLocaleString(locale, 'f', -128)){
-                        return root.swapInputParamsForm.fromTokenAmount
```

Comment:
This needs to go away, creates an endless binding loop between `tokenAmount` and `root.swapInputParamsForm.fromTokenAmount`

Reply:
Ive added a function to do that here https://github.com/status-im/status-desktop/pull/15261/files#:~:text=function%20updateInputText()%20%7B

Reply:
I'll take a look

Reply:
No, it doesn't work

Reply:
In the end, I settled on:
```qml
tokenAmount: root.swapInputParamsForm.fromTokenAmount
``` which works fine

Reply:
To explain:
```qml
SQUtils.AmountsArithmetic.fromString(value)
```
could have never worked since `value` is a double and:

```
.toLocaleString()
```
can't work on the BigInt; doesn't exist